test/gtest/ucp/ucp_test.cc

```diff
@@ -50,9 +46,24 @@ void ucp_test::cleanup() {
     m_entities.clear();
 }
 
-ucp_test::entity* ucp_test::create_entity() {
+void ucp_test::init() {
```

Comment:
"another class - ucp_p2p_test"
This is completely different design. Current idea is to create common class that can be used in UCP and UCT tests (including common entity access functions sender()/receiver()). UC[P|T] test class trees are quite different and the only place where we can use common code (for `entity` handling) is in a base class for `class uc[p|t]_test`.
Common functions sender()/receiver() planned to be used in both UC[P|T] tests.

As I understand you propose similar design as it implemented in UCT (class uct_p2p_test). In this approach we have entity storage in base class (uct_test) but access functions in derived class `uct_p2p_test`. In this case I can't see how we can reuse entity handling code for both UC[P|T] tests tree.